sqlserver/datadog_checks/sqlserver/activity.py

```diff
@@ -68,7 +73,8 @@
     INNER JOIN sys.dm_exec_requests req
         ON c.connection_id = req.connection_id
     CROSS APPLY sys.dm_exec_sql_text(req.sql_handle) qt
-WHERE sess.session_id != @@spid AND sess.status != 'sleeping'
+WHERE sess.session_id != @@spid
+    AND (sess.status != 'sleeping' OR sess.session_id IN (SELECT blocking_session_id FROM cteBlocking) OR isnull(req.blocking_session_id, 0) <> 0)
```

Comment:
I think if we looked in the active transactions table this might be a tad bit faster (would be a smaller DMV table) 

if a thing is blocking, in a `sleeping` state I believe it should always show up there

Reply:
But would it be possible to have a blocking session not in a transaction? Not sure those would be functionally equivalent. 

But I do believe in the idea that we should collect all processes with an active transaction; that seems like a better improvement overall.

Reply:
A subquery to capture idle blockers could slow down the query and might be an impediment for increasing the sampling frequency. If we assume that in most samples we won't find any blockers a better approach could be: while scanning the result set of only active sessions build a set containing blocker `spids`. In the end, execute a second query to collect only the most basic information on idle blockers - as they aren't executing anything we could probably omit all the joins. On the downside, this would generate an additional round trip to the database in the case some blockers are found, but I think this is a good trade off, as we're optimizing for the most frequent scenario where there aren't idle blockers.